extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -270,7 +270,8 @@ public void defineHealthRoutes(BuildProducer<RouteBuildItem> routes,
     public void processSmallRyeHealthConfigValues(SmallRyeHealthBuildTimeConfig healthConfig,
             BuildProducer<RunTimeConfigurationDefaultBuildItem> config) {
         if (healthConfig.contextPropagation()) {
-            config.produce(new RunTimeConfigurationDefaultBuildItem("io.smallrye.health.context.propagation", "true"));
+            LOG.warn(
+                    "The config property quarkus.smallrye-health.context-propagation is deprecated and will be removed. The context is now always propagated.");
```

Comment:
Should we mark the property as `@Deprecated(since = "3.30`, forRemoval = true)`?